art/attacks/evasion/overload.py

```diff
@@ -0,0 +1,260 @@
+            for ii in range(self.num_grid):
+                for jj in range(self.num_grid):
+                    x1 = ii * stride_x
+                    y1 = jj * stride_y
+                    x2 = min(x1 + stride_x, x.shape[-2])
+                    y2 = min(y1 + stride_y, x.shape[-1])
+                    bb = torch.as_tensor([x1, y1, x2, y2], device=x.device)[None, :]
+                    grid_box = torch.cat([grid_box, bb], dim=0)
+
+            for xi in range(x.shape[0]):
+                xyhw = adv_logits[xi, :, :4]
+                prob = torch.max(adv_logits[xi, :, 5:], dim=1).values
+                box_idx = adv_logits[xi, :, 4] * prob > THRESHOLD
+                xyhw = xyhw[box_idx]
+                c_xyxy = self.xywh2xyxy(xyhw)
+                scores = self.box_iou(grid_box, c_xyxy)
+                scores = torch.where(scores > 0.0, torch.ones_like(scores), torch.zeros_like(scores))
+                scores = torch.sum(scores, dim=1)
+
+                idx_min = torch.argmin(scores)
+                grid_min = grid_box[idx_min]
+                x1, y1, x2, y2 = grid_min.int()
+                pixel_weight = pixel_weight / torch.max(pixel_weight[xi,:]) / 255.0
+
+        return ind_loss, pixel_weight
+
+    def xywh2xyxy(self, xywh: "torch.tensor") -> "torch.tensor":
+        """
+        Convert the representation from xywh format yo xyxy format.
+        : param xyhw: A n by 4 boxes store the information in xyhw format
+                      where [x ,y, w h] is [center_x, center_y, width, height]
+        : return: The n by 4 boxex in xyxy format
+                  where [x1, y1, x2, y2] is [top_left_x, top_left_y, bottom_right_x,  bottom_right_y]
+        """
+        xyxy = xywh.clone()
+        xyxy[:, 0] = xywh[:, 0] - xywh[:, 2] / 2
+        xyxy[:, 1] = xywh[:, 1] - xywh[:, 3] / 2
+        xyxy[:, 2] = xywh[:, 0] + xywh[:, 2] / 2
+        xyxy[:, 3] = xywh[:, 1] + xywh[:, 3] / 2
+        return xyxy
+
+    def box_iou(self, box1: "torch.tensor", box2: "torch.tensor", eps: float = 1e-7) -> "torch.tensor":
+        """
+        Return intersection-over-union (Jaccard index) of boxes.
+        Both sets of boxes are expected to be in (x1, y1, x2, y2) format.
+        Arguments:
+            box1 (Tensor[N, 4])
+            box2 (Tensor[M, 4])
+        Returns:
+            iou (Tensor[N, M]): the NxM matrix containing the pairwise
+                IoU values for every element in boxes1 and boxes2
+        # https://github.com/pytorch/vision/blob/master/torchvision/ops/boxes.py
+        """
+        import torch
+
+        # inter(N,M) = (rb(N,M,2) - lt(N,M,2)).clamp(0).prod(2)
+        (a1, a2), (b1, b2) = box1.unsqueeze(1).chunk(2, 2), box2.unsqueeze(0).chunk(2, 2)
+        inter = (torch.min(a2, b2) - torch.max(a1, b1)).clamp(0).prod(2)
+
+        # IoU = inter / (area1 + area2 - inter)
+        return inter / ((a2 - a1).prod(2) + (b2 - b1).prod(2) - inter + eps)
+
+    def _check_params(self) -> None:
```

Comment:
I think it would be very helpful if we could add a check if the provided model is a Yolo v5 model.